tests/sentry/filters/test_legacy_browsers.py

```diff
@@ -1,7 +1,171 @@
+    'opera_11': 'Opera/9.80 (Windows NT 5.1; U; it) Presto/2.7.62 Version/11.00',
+    'opera_12': 'Opera/9.80 (X11; Linux i686; Ubuntu/14.10) Presto/2.12.388 Version/12.16',
+    'opera_15': 'Mozilla/5.0 (X11; Linux x86_64; Debian) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/28.0.1500.52 Safari/537.36 OPR/15.0.1147.100',
+    'chrome': 'Mozilla/5.0 (Windows NT 6.1) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/41.0.2228.0 Safari/537.36',
+    'edge': 'Mozilla/5.0 (Windows NT 10.0) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/42.0.2311.135 Safari/537.36 Edge/12.10136',
+    'safari_5': 'Mozilla/5.0 (Windows; U; Windows NT 6.1; zh-HK) AppleWebKit/533.18.1 (KHTML, like Gecko) Version/5.0.2 Safari/533.18.5',
+    'safari_7': 'Mozilla/5.0 (Macintosh; Intel Mac OS X 10_9_3) AppleWebKit/537.75.14 (KHTML, like Gecko) Version/7.0.3 Safari/7046A194A',
+}
+
+
+class SetLegacyBrowserFilterTest(APITestCase):
+
+    def test_set_default_all_browsers(self):
+        self.login_as(user=self.user)
+        project = self.create_project()
+
+        url = reverse('sentry-api-0-project-filters', kwargs={
+            'organization_slug': project.organization.slug,
+            'project_slug': project.slug,
+            'filter_id': "legacy-browsers"
+        })
+        response = self.client.put(url, data={
+            'active': True,
+        })
+        assert response.status_code == 201, response.content
+
+        options = ProjectOption.objects.get_value(
+            project=project,
+            key='filters:legacy-browsers'
+        )
+        assert options == '1'
+
+    def test_set_default_no_browsers(self):
+        self.login_as(user=self.user)
+        project = self.create_project()
+
+        url = reverse('sentry-api-0-project-filters', kwargs={
+            'organization_slug': project.organization.slug,
+            'project_slug': project.slug,
+            'filter_id': "legacy-browsers"
+        })
+        response = self.client.put(url, data={
+            'active': False,
+        })
+        assert response.status_code == 201, response.content
+
+        options = ProjectOption.objects.get_value(
+            project=project,
+            key='filters:legacy-browsers'
+        )
+        assert options == '0'
+
+    def test_set_opera(self):
+        self.login_as(user=self.user)
+        project = self.create_project()
+
+        url = reverse('sentry-api-0-project-filters', kwargs={
+            'organization_slug': project.organization.slug,
+            'project_slug': project.slug,
+            'filter_id': "legacy-browsers"
+        })
+        response = self.client.put(url, data={
+            'subfilters': ["opera"]
+        })
+        assert response.status_code == 201, response.content
+
+        options = ProjectOption.objects.get_value(
+            project=project,
+            key='filters:legacy-browsers'
+        )
+        assert options == set(['opera'])
```

Comment:
Not a big deal here at all, just an FYI, `set` has shorter syntax, so you can do `{'opera'}` which is the same thing. or `{'a', 'b'}`, etc.